modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/JsonProcessorTests.java

```diff
@@ -227,11 +230,85 @@ public void testAddToRootNonRecursiveMerge() throws Exception {
         assertThat(exception.getMessage(), containsString("cannot add non-map fields to root of document"));
     }
+
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    public void testApply() {
+        {
+            Object result = JsonProcessor.apply("{\"foo\":\"bar\"}", true, true);
+            assertThat(result, instanceOf(Map.class));
+            Map resultMap = (Map) result;
+            assertThat(resultMap.size(), equalTo(1));
+            assertThat(resultMap.get("foo"), equalTo("bar"));
+        }
+        {
+            Object result = JsonProcessor.apply("\"foo\"", true, true);
+            assertThat(result, instanceOf(String.class));
+            assertThat(result, equalTo("foo"));
+        }
+        {
+            boolean boolValue = randomBoolean();
+            Object result = JsonProcessor.apply(Boolean.toString(boolValue), true, true);
+            assertThat(result, instanceOf(Boolean.class));
+            assertThat(result, equalTo(boolValue));
+        }
+        {
+            double value = randomDouble();
+            Object result = JsonProcessor.apply(Double.toString(value), true, true);
+            assertThat(result, instanceOf(Double.class));
+            assertThat((double) result, closeTo(value, .001));
+        }
+        {
+            List<Double> list = randomList(10, ESTestCase::randomDouble);
+            String value = list.stream().map(val -> Double.toString(val)).collect(Collectors.joining(",", "[", "]"));
+            Object result = JsonProcessor.apply(value, true, true);
+            assertThat(result, instanceOf(List.class));
+            List<Double> resultList = (List<Double>) result;
+            assertThat(resultList.size(), equalTo(list.size()));
+            for (int i = 0; i < list.size(); i++) {
+                assertThat(resultList.get(i), closeTo(list.get(i), .001));
+            }
+        }
+        expectThrows(IllegalArgumentException.class, () -> JsonProcessor.apply("foo", true, true));
+        expectThrows(IllegalArgumentException.class, () -> JsonProcessor.apply("foo", true, false));
+        expectThrows(IllegalArgumentException.class, () -> JsonProcessor.apply("foo [360113.865822] wbrdg-0afe001ce", true, true));
+        expectThrows(IllegalArgumentException.class, () -> JsonProcessor.apply("foo [360113.865822] wbrdg-0afe001ce", true, false));
+
+        /*
+         * The following are examples of malformed json. Previously apply parsed just the first token and ignored the rest, but it now
+         * throw an IllegalArgumentException unless strictJsonParsing is set to false. See
+         * https://github.com/elastic/elasticsearch/issues/92898.
+         */
+        expectThrows(IllegalArgumentException.class, () -> JsonProcessor.apply("123 foo", true, true));
+        expectThrows(IllegalArgumentException.class, () -> JsonProcessor.apply("45 this is {\"a\": \"json\"}", true, true));
+
+        {
```

Comment:
These are all good. Maybe split into a separate test just to make it clear that we're testing for something different than the code above? Basically, maybe the "strict validation" stuff could be its own test.